src/controller/java/templates/partials/decode_value.zapt

```diff
@@ -60,7 +60,7 @@ if ({{source}}.IsNull()) {
         env->SetByteArrayRegion({{target}}ByteArray, 0, static_cast<jsize>({{source}}.size()), reinterpret_cast<const jbyte *>({{source}}.data()));
         {{target}} = {{target}}ByteArray;
       {{else if (isCharString type)}}
-        {{target}} = env->NewStringUTF(std::string({{source}}.data(), {{source}}.size()).c_str());
+        {{target}} = chip::JniReferences::GetInstance().CharToJniString(std::string({{source}}.data(), {{source}}.size()).c_str());
```

Comment:
In theory UTF8 should be valid for string, because this is not octet string. Switching this from UTF8 does not seem correct.

Reply:
```
7.18.2.28. Character String
The character string data type is derived from an octet string. The octets SHALL be characters with UTF-8 encoding. An instance of this data type SHALL NOT contain truncated code points.
```

We should report an exception (can we add tests here for fast return?) but not accept JNI strings that are not UTF8.

Reply:
I totally agree with you that it should be valid for a string. However, it is also necessary that a crash should not occur when there is a value that is not vailed. By any chance, is there a better solution? The solution I suggested is handled by UTF-8, but it's a way to change it in a different way so that it doesn't cause a crash.

Reply:
Note that `NewStringUTF` does not actually accept standard UTF-8.  If you have a standard UTF-8 encoding of an emoji and you try to `NewStringUTF` with it, I expect it fails.

Reply:
Is it the case that NewStringUTF does not accept valid strings (if so that is a bug and we should fix it) or that this is indeed an invalid string (this is my current understanding).

For an invalid string, I expect java to throw an exception on decoding.

Reply:
My expectation:
   - UTF8 validation should be ok. We should replace NewStringUTF with something else if we prove a valid emoji (in UTF-8 encoding) fails to parse. I don't think this should be the case, however java docs mentions `Constructs a new java.lang.String object from an array of characters in modified UTF-8 encoding.` ... what is `modified` ??
    - we should not crash, however throwing an exception should be acceptable. Maybe we need to check `ExceptionOcurred` in our decoders.

@bzbarsky-apple do you have a concrete example with the emoji?

Reply:
For what its worth, the string described in this issue does look like invalid UTF-8:

```python
>>> binascii.a2b_hex('74666d5f626173d5').decode('utf-8')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xd5 in position 7: unexpected end of data
```

Reply:
Maybe it is truncated because d5 is a 2-byte continuation start, so something like `74666d5f626173d5a0` would be valid utf8.

Reply:
Regarding "modified" the docs seem to say things that should not affect us:
  - nulls are encoded as 2-bytes (we should not care ...we should not embed nulls)
  - only support 1,2 or 3 byte encoding. I do not believe we have a reason to use 4-byte utf8 in chip

Reply:
@andy31415 @bzbarsky-apple 
It has been modified to return as an error when it is not decoded as `UTF-8`.
(In a file created with zap, it will be returned as null.)